rsocket-micrometer/src/main/java/io/rsocket/micrometer/MicrometerDuplexConnection.java

```diff
@@ -0,0 +1,244 @@
+
+    private final Counter keepalive;
+
+    private final Counter lease;
+
+    private final Counter metadataPush;
+
+    private final Counter next;
+
+    private final Counter nextComplete;
+
+    private final Counter requestChannel;
+
+    private final Counter requestFireAndForget;
+
+    private final Counter requestN;
+
+    private final Counter requestResponse;
+
+    private final Counter requestStream;
+
+    private final Counter resume;
+
+    private final Counter resumeOk;
+
+    private final Counter setup;
+
+    private FrameCounters(Type connectionType, MeterRegistry meterRegistry, Tag... tags) {
+      this.cancel = counter(connectionType, meterRegistry, CANCEL, tags);
+      this.complete = counter(connectionType, meterRegistry, COMPLETE, tags);
+      this.error = counter(connectionType, meterRegistry, ERROR, tags);
+      this.extension = counter(connectionType, meterRegistry, EXT, tags);
+      this.keepalive = counter(connectionType, meterRegistry, KEEPALIVE, tags);
+      this.lease = counter(connectionType, meterRegistry, LEASE, tags);
+      this.metadataPush = counter(connectionType, meterRegistry, METADATA_PUSH, tags);
+      this.next = counter(connectionType, meterRegistry, NEXT, tags);
+      this.nextComplete = counter(connectionType, meterRegistry, NEXT_COMPLETE, tags);
+      this.requestChannel = counter(connectionType, meterRegistry, REQUEST_CHANNEL, tags);
+      this.requestFireAndForget = counter(connectionType, meterRegistry, REQUEST_FNF, tags);
+      this.requestN = counter(connectionType, meterRegistry, REQUEST_N, tags);
+      this.requestResponse = counter(connectionType, meterRegistry, REQUEST_RESPONSE, tags);
+      this.requestStream = counter(connectionType, meterRegistry, REQUEST_STREAM, tags);
+      this.resume = counter(connectionType, meterRegistry, RESUME, tags);
+      this.resumeOk = counter(connectionType, meterRegistry, RESUME_OK, tags);
+      this.setup = counter(connectionType, meterRegistry, SETUP, tags);
+    }
+
+    @Override
+    public void accept(Frame frame) {
+      FrameType frameType = frame.getType();
+
+      switch (frameType) {
+        case RESERVED:
+          break;
+        case SETUP:
+          this.setup.increment();
+          break;
+        case LEASE:
+          this.lease.increment();
+          break;
+        case KEEPALIVE:
+          this.keepalive.increment();
+          break;
+        case REQUEST_RESPONSE:
+          this.requestResponse.increment();
+          break;
+        case REQUEST_FNF:
+          this.requestFireAndForget.increment();
+          break;
+        case REQUEST_STREAM:
+          this.requestStream.increment();
+          break;
+        case REQUEST_CHANNEL:
+          this.requestChannel.increment();
+          break;
+        case REQUEST_N:
+          this.requestN.increment();
+          break;
+        case CANCEL:
+          this.cancel.increment();
+          break;
+        case PAYLOAD:
```

Comment:
It seems like a mistake to not track payload frames.

Reply:
The oversight here was actually including the `enum` and doing nothing with it (ah, IDE's conveniently adding everything).  Experimentally I noticed that `PAYLOAD` frames are never transferred, instead being immediately translated into [`COMPLETE`, `NEXT` and `NEXT_COMPLETE` frames](https://github.com/rsocket/rsocket-java/blob/1.0.x/rsocket-core/src/main/java/io/rsocket/framing/FrameType.java#L193-L203).  I'll add in counter for `PAYLOAD` anyway (since it can't hurt), but I suspect it'll always be 0.